Gemfile

```diff
@@ -12,6 +12,11 @@ gem 'facter', '>= 3.0', '!= 4.0.52'
 gem 'puppet-strings'
 gem 'rake'
 
+if RUBY_VERSION >= '3.4'
+  gem 'base64'
+  gem 'getoptlong'
```

Comment:
Did you figure out why we need this? We should make sure that dependency pulls it in.

Reply:
Puppet does -- https://github.com/puppetlabs/puppet/blob/e227c27540975c25aa22d533a52424a9d2fc886a/puppet.gemspec#L27

Reply:
That already lists and we list `puppet` so why do we need to list it again?

Reply:
Sorry, I linked to the wrong version, 7.34.0 is used in SRPM build and you will note this is missing there:

https://github.com/puppetlabs/puppet/blob/7.34.0/puppet.gemspec

Reply:
Should we move to using Puppet 8 instead of trying to fix it with Puppet 7? I'm a bit worried about other dragons hiding in there.

Reply:
That thought crossed my mind. I thought maybe we should get builds back on track with 7 and then investigate puppet 8 / openvox.